modules/reactor/src/meshgenerators/AssemblyMeshGenerator.C

```diff
@@ -384,121 +420,124 @@ AssemblyMeshGenerator::AssemblyMeshGenerator(const InputParameters & parameters)
   }
   else
     declareMeshProperty("extruded", false);
+
+  // Save updates to name id map to ReactorMeshParams object
+  updateReactorMeshParams();
 }
 
 std::unique_ptr<MeshBase>
 AssemblyMeshGenerator::generate()
 {
+  // Re-initialize name_id_map and current_block_id from ReactorMeshParams object
+  // in case variables have been modified between constructor and this method
+  initializeReactorMeshParams();
+
   // This generate() method will be called once the subgenerators that we depend on are
-  // called. This is where we set the region ids and element integers.
+  // called. This is where we set all element integers.
 
+  // Define all extra element names and integers
   std::string plane_id_name = "plane_id";
   std::string region_id_name = "region_id";
   std::string pin_type_id_name = "pin_type_id";
+  std::string assembly_type_id_name = "assembly_type_id";
 
   if (!(*_build_mesh)->has_elem_integer(region_id_name) ||
       !(*_build_mesh)->has_elem_integer(pin_type_id_name))
-    mooseError("Expected mesh inputs to have extra integer ids");
+    mooseError("Expected mesh inputs to have region_id and pin_type_id extra integer ids");
 
   unsigned int ptid_int = (*_build_mesh)->get_elem_integer_index(pin_type_id_name);
   unsigned int rid_int = (*_build_mesh)->get_elem_integer_index(region_id_name);
 
-  // go through the background and duct regions and assign their elements
-  // pin_type id placeholders and region id place holders
-  for (const auto i : make_range(_peripheral_regions))
-  {
-    subdomain_id_type region = UINT16_MAX - (i + 1);
-
-    for (const auto & elem : as_range((*_build_mesh)->active_subdomain_elements_begin(region),
-                                      (*_build_mesh)->active_subdomain_elements_end(region)))
-    {
-      elem->set_extra_integer(ptid_int, region); // identifier for element not in a pin
-      elem->set_extra_integer(rid_int, region);
-    }
-  }
+  unsigned int atid_int, pid_int = 0;
+  if (!(*_build_mesh)->has_elem_integer(assembly_type_id_name))
+    atid_int = (*_build_mesh)->add_elem_integer(assembly_type_id_name);
+  else
+    atid_int = (*_build_mesh)->get_elem_integer_index(assembly_type_id_name);
 
   if (_extrude)
   {
-    // swap the region ids on the subdomain ids to the correct ones
-    // for their axial layer
     if (!(*_build_mesh)->has_elem_integer(plane_id_name))
-      mooseError("Expected extruded mesh to have plane IDs");
+      mooseError("Expected extruded mesh to have plane_id extra integers");
+    pid_int = (*_build_mesh)->get_elem_integer_index(plane_id_name);
```

Comment:
you do this thing 3 times, let's have a routine

Reply:
Created the function `getElemIntegerFromMesh`